rerun_js/web-viewer/index.js

```diff
@@ -131,4 +131,16 @@ export class WebViewer {
     this.#canvas = null;
     this.#handle = null;
   }
+
+  /**
+   * Add an rrd to the viewer from a byte array.
+   *
+   * @param {Uint8Array} rrd stored in a byte array, received via some other side channel.
+   */
+  add_rrd_from_bytes(rrd) {
```

Comment:
Note: We can have this discussion later, and merge this PR in the meantime to unblock follow-up work

Should we instead expose an API for opening _streams_ to which log messages can be pushed? A stream could have a `close` method.

```js
class WebViewer {
  /**
   * Creates a new log message stream.
   *
   * The stream can be used to incrementally push `rrd` log messages into the viewer.
   * 
   * @returns {LogStream}
   */
  stream() {
    if (!this.#handle) throw new Error("...");
    const id = crypto.randomUUID();
    this.#handle.open_channel(id);
    const on_send = (data) => this.#handle.push_data_to_channel(id, data);
    const on_close = () => this.#handle.close_channel(id);
    const get_state = () => this.#state;
    return new LogStream(on_send, on_close, get_state);
  }
}

class LogStream {
  #on_send;
  #on_close;
  #get_state;
  #closed = false;
  
  /** @private */
  constructor(on_send, on_close, get_state) {
    this.#on_send = on_send;
    this.#on_close = on_close;
    this.#get_state = get_state;
  }
  
  get ready() {
    return !this.#closed && this.#get_state() === "ready";
  }
  
  /**
   * Send an `rrd` log message to the viewer.
   *
   * Does nothing if `!this.ready`.
   */
  push(data) {
    if (!this.ready) return;
    this.#on_send(data);
  }

  /**
   * Close the stream.
   *
   * Does nothing if `!this.ready`.
   */
  close() {
    if (!this.ready) return;
    this.#on_close();
    this.#closed = true;
  }
}
```